### PR TITLE
feat(*): add handling for issue comment events

### PR DESCRIPTION
**Pull request by vdice**
Adds handling of webhooks with event type `issue_comment`.

When such an event comes through, and the issue comment corresponds to a pull request, we will fetch said pull request data (commit, branch, etc.) as well as provide an installation token on the resulting Brigade build, so that users can trigger check runs (in addition to other actions) from said issue comment event.  (Also note, we only add PR data and token _if_ the author association on the issue comment is in the `allowedAuthors` list setup on the GitHub App gateway.)

In doing so, I also added the ability to provide repo name, commit and branch to the check run process, to take precedence over attempted derivation from the event payload.  This was done as the payload type may not be from the Checks API, but we may still desire a check run to be triggered.

This logic was tested with the example as seen/added in `docs/examples`, after updating my Brigade GitHub App image (and check run image in the `brigade.js` file) to use tags built from this branch.

Depending on what form of this change set goes in, I'm eager to follow up with a refactor PR, mostly centered around DRY-ing up logic in the `github.go` file (such as adding helpers around installation token generation, project retrieval + signature/secret verification and others).

### CODE REVIEW

File: pkg/webhook/github.go [hunk 1]
```diff
@@ -240,15 +425,23 @@ func (s *githubHook) handleEvent(c *gin.Context, eventType string) {
 	var action string
 
 	switch e := e.(type) {
-	case *github.PushEvent:
-		// If this is a branch deletion, skip the build.
-		if e.GetDeleted() {
-			c.JSON(http.StatusOK, gin.H{"status": "build skipped on branch deletion"})
-			return
-		}
+	case *github.CommitCommentEvent:
```
Comment (vdice) [inline pkg/webhook/github.go]: (Diffs in this section should just be from ordering alphabetically...)

**Comment by krancour** [comment]
@vdice, I don't see any obvious problems, but I'm also not 100% understanding some of the nuances here. Any chance you can give me a brief walkthrough on Monday?

**Comment by vdice** [comment]
@krancour definitely.  It'll be a great way to see what clarity/docs/changes we can make to help other reviewers as well.

**Comment by krancour** [comment]
Trying to understand the changes to the check-run image. Pending some better understanding of that, this LGTM.

**Comment by krancour** [comment]
> Trying to understand the changes to the check-run image. Pending some better understanding of that, this LGTM.

Nevermind. I understand the purpose of those changes now. It's to allow certain data to be provided when creating the job _instead of_ allowing that data to be extracted from the payload-- which might not be the right _type_ of payload that that container expects.

I'm a _little bit_ hesitant on this one aspect of things. By having that container parse the payload for actual check_run:* events we are processing, but taking a different path and seeding that container with extra env vars (to bypass that parsing) when handling an issue_comment event, are we promoting an inconsistency that we could avoid by, perhaps, making that check-run container understand how to parse _both_ of the possible payload types?

(It's entirely possible that there's a technical barrier to this approach that I have failed to notice.)

**Comment by vdice** [comment]
> I'm a little bit hesitant on this one aspect of things. By having that container parse the payload for actual check_run:* events we are processing, but taking a different path and seeding that container with extra env vars (to bypass that parsing) when handling an issue_comment event, are we promoting an inconsistency that we could avoid by, perhaps, making that check-run container understand how to parse both of the possible payload types?

Yes, good questions/thoughts.  It would be preferable to just add this new event type (`issue_comment`) to the list as a parsable payload and do away with the env var overrides, but, alas, this [event type](https://developer.github.com/v3/activity/events/types/#issuecommentevent) doesn't have commit or branch fields that we'd need to be able to set/get.  (I'm also not seeing an area of the payload for this event type where we might add user-defined data for this case.)

I'll keep looking for a bit to see if there's a better approach...

**Comment by vdice** [comment]
@technosophos @krancour I've revisited the check run logic to remove the added env vars and instead update Brigade's `webhook.Payload` object with `Commit` and `Branch` fields, which the check run utility can then access for requesting a check run when the event is of type `issue_comment`.  See the [latest commit](https://github.com/brigadecore/brigade-github-app/pull/56/commits/f34e783fe64fa3650edd4bc0d730ec16704406f2)

Let me know what you think of this approach.  We could entertain the notion of always populating these new fields and then deferring to these values in the check run utility for all supported event types, though, if that gains momentum, I'd prefer to include in a follow-up refactor...

**Comment by krancour** [comment]
@vdice I like the changes in your latest commit. Will let @technosophos sound off as well before LGTM'ing, but I think this is 👌 .

**Pull request merged**